add(platform): backup to another region (BTAR) (#6)
aiven/aiven-docs: docs/platform/howto/btar/manage-backup-to-another-region.md

:::

## Fork and restore a service with BTAR {#fork-and-restore}

You can use the [Aiven Console](https://console.aiven.io/) to recover your service from a
backup in another region. To restore your service using BTAR,
[create a fork](/docs/platform/howto/console-fork-service) of the original service in the
region where the secondary backup resides.

:::note
When you **Fork & restore** from the secondary backup, your new fork service is created in
the cloud and region where the secondary backup is located. The fork service gets the same
plan that the primary service uses. Backups of the fork service are located in the region
where this new service is hosted.
:::

1. Open the [Aiven Console](https://console.aiven.io/) and navigate to your service
   homepage.
1. Select **Backups** from the sidebar.
1. On the **Backups** page, select **Fork & restore**.
1. In the **New database fork** window, apply the following settings:

   1. As **Source backup location**, select **Secondary location**.
   1. As **Source backup version**, select either **Latest transaction** or **Point in
      time**.

      :::note

>       For the point-in-time recovery (PITR) option, set up the time to no later than the

Author: Does this mean that for GA we don't select "source backup verions" in the console? Or is this option going to show with the preselected "Latest transation" or how else user experience is going to look like?

>       For the point-in-time recovery (PITR) option, set up the time to no later than the

Author: In that case, I'm commenting out this option as no user action is needed. I'll bring it back when we actually enable it post GA.

---
title: Manage a cross-region backup
sidebar_label: Manage cross-region backup
pro: true
---

For a service that has the [backup to another region (BTAR) feature](/docs/platform/concepts/backup-to-another-region) enabled, you can check the service backup status, change the backup region, monitor the replication lag, fork and restore using [the cross-region backup](/docs/platform/concepts/backup-to-another-region), or migrate to another cloud or region.

:::important
Backup to another region (BTAR) is available on

> `[Pro Platform](/docs/platform/concepts/pro-platform)`.

Reviewer: ⚠️

pro: true
---

For a service that has the [backup to another region (BTAR) feature](/docs/platform/concepts/backup-to-another-region) enabled, you can check the service backup status, change the backup region, monitor the replication lag, fork and restore using [the cross-region backup](/docs/platform/concepts/backup-to-another-region), or migrate to another cloud or region.

:::important
Backup to another region (BTAR) is available on
`[Pro Platform](/docs/platform/concepts/pro-platform)`.
:::

## Prerequisites

You have at least one
[Aiven service with BTAR enabled](/docs/platform/howto/btar/enable-backup-to-another-region).

## Change a backup region

1. Log in to the [Aiven Console](https://console.aiven.io/).
1. From the **Services** view, select an Aiven service on which you'd like to enable BTAR.
1. On your service's page, select **Backups** from the sidebar.
1. On the **Backups** page, select the actions (**...**) menu > **Secondary backup
   location**.
1. In the **Edit secondary backup location** window, use the **Secondary location**
   dropdown menu to select a region for your additional backup. Confirm your choice by
   selecting **Save**.

:::important

> When you change the backup region, a new base backup is triggered. To change the backup

Author: @ali-gholami-aiven , could you please re-review this note here? I've modified it to make it more specific and less ambiguous.

pro: true
---

For a service that has the [backup to another region (BTAR) feature](/docs/platform/concepts/backup-to-another-region) enabled, you can check the service backup status, change the backup region, monitor the replication lag, fork and restore using [the cross-region backup](/docs/platform/concepts/backup-to-another-region), or migrate to another cloud or region.

:::important
Backup to another region (BTAR) is available on
`[Pro Platform](/docs/platform/concepts/pro-platform)`.
:::

## Prerequisites

You have at least one
[Aiven service with BTAR enabled](/docs/platform/howto/btar/enable-backup-to-another-region).

## Change a backup region

1. Log in to the [Aiven Console](https://console.aiven.io/).
1. From the **Services** view, select an Aiven service on which you'd like to enable BTAR.
1. On your service's page, select **Backups** from the sidebar.
1. On the **Backups** page, select the actions (**...**) menu > **Secondary backup
   location**.
1. In the **Edit secondary backup location** window, use the **Secondary location**
   dropdown menu to select a region for your additional backup. Confirm your choice by
   selecting **Save**.

:::important

> Think carefully before you decide to change a backup region. Since each change of a backup
> region triggers a new base backup, you can change a backup region only infrequently.

Reviewer: Let's not pretend readers don't already think carefully. What do you mean by infrequently? What's the consequence for me as a user, I can't switch back in the next 24h? 7days? Until the backup's over?

Consider something like this that's shorter and doesn't generate more questions?

```suggestion
Changing a backup region triggers a new base backup that can take up to X minutes. You cannot switch back until the backup is complete.
```

:::

## Monitor a service with BTAR

> There are a few things you may want to check for your Aiven service in the context for
> BTAR:

Reviewer: ```suggestion
There are a few things you may want to check for your Aiven service in the context of
BTAR:
```

- What is the status of a secondary backup?

  - Does your service has a backup in another region?
  - What is the target region of the secondary backup?

- What is the replication lag between data availability in the primary region and the
  secondary region?

### Check BTAR status in the console

To check out the availability, the status, and the target region of a secondary (BTAR)
backup in the [Aiven Console](https://console.aiven.io/), navigate to your
**service's page** > **Backups** view > **Secondary backup location** column.

### Check replication lag with API

> To check out the target region and the replication lag for a secondary (BTAR) backup of

Reviewer: Nit: It's `check` in the title, `check out` in the text (which is rather informal). What about `determine` to avoid phrasal verbs and ensure uniformity?

your service, call the
[ServiceBackupToAnotherRegionReport](https://api.aiven.io/doc/#tag/Service/operation/ServiceBackupToAnotherRegionReport)
endpoint.

Configure the call as follows:

1. Enter YOUR-PROJECT-NAME and YOUR-SERVICE-NAME into the URL.
1. Specify DESIRED-TIME-PERIOD depending on the time period you need the metrics for:
   select one of the following values for the ``period`` key: ``hour``, ``day``, ``week``,
   ``month``, or ``year``.

```bash
    curl --request POST                                                                                                      \
        --url https://api.aiven.io/v1/project/YOUR-PROJECT-NAME/service/YOUR-SERVICE-NAME/backup_to_another_region/report    \
        --header 'Authorization: Bearer YOUR-BEARER-TOKEN'                                                                   \
        --header 'content-type: application/json'                                                                            \
        --data '{"period":"DESIRED-TIME-PERIOD"}'
```

> :::note[Output]
> As an output, you get metrics including replication lags at specific points in time.
> :::

Reviewer: ```suggestion
As output, you get metrics including replication lags at specific points in time.
```
